src/zoid/buttons/component.jsx

```diff
@@ -681,6 +682,51 @@ export const getButtonsComponent: () => ButtonsComponent = memoize(() => {
         },
       },
 
+      onMessageHover: {
+        type: "function",
+        required: false,
+        value: ({ props }) => {
+          return () => {
+            // lazy loads the modal, to be memoized and executed onMessageClick
+            const { clientID, merchantID } = props;
+            return getModal(clientID, merchantID);
+          };
+        },
+      },
+
+      onMessageClick: {
+        type: "function",
+        required: false,
+        value: ({ props }) => {
+          return async ({ offerType, messageType }) => {
+            const { message, clientID, merchantID, currency, buttonSessionID } =
+              props;
+            const amount = message?.amount || undefined;
+
+            const modalInstance = await getModal(clientID, merchantID);
+
+            getLogger()
+              .info("button_message_clicked")
+              .track({
+                [FPTI_KEY.EVENT_NAME]: "message_click",
+                // [FPTI_KEY.BUTTON_MESSAGE_OFFER_TYPE]: offerType,
+                "button message type fpti key placeholder": messageType,
+                // [FPTI_KEY.BUTTON_MESSAGE_POSITION]: message.position,
+                // [FPTI_KEY.BUTTON_MESSAGE_ALIGN]: message.align,
+                // [FPTI_KEY.BUTTON_MESSAGE_COLOR]: message.color,
+                // [FPTI_KEY.AMOUNT]: amount,
```

Comment:
do we still need the commented out keys?

Reply:
Sorry, I jumped the gun a bit on hitting the 'ready for review' button after I finished the feature and tests but before merging the other PR.  These commented-out keys will be replaced very soon with updated, more precise keys from [a more dedicated tracking PR](https://github.com/paypal/paypal-checkout-components/pull/2360/files#diff-30a215bcfe3877b1f55af82ba98b729149bab885b3c5d2921d3bdc7e1c9ac799R748-R768).  Will let you know again when that PR is merged.